CRVResponse/src/CrvStepsFromStepPointMCs_module.cc

```diff
@@ -0,0 +1,361 @@
+	fhicl::Atom<int> diag{ Name("diagLevel"),
+	  Comment("Diag Level"), 0};
+	fhicl::Atom<unsigned> csize{ Name("OutputCollectionSize"),
+	  Comment("Estimated size of output collection"), 2000};
+	fhicl::Atom<string> crvSteps { Name("crvStepPoints"),
+	  Comment("Tracker StepPointMC Instance name"),"CRV"};
+	fhicl::Atom<string> stepsToSkip { Name("SkipTheseStepPoints"),
+	  Comment("Tracker StepPointMC producers to skip"), ""};
+	fhicl::Atom<unsigned> startSize { Name("StartSize"),
+	  Comment("Starting size for crvIndex-particle vector"),4};
+	fhicl::Atom<bool> removeNeutralParticles{ Name("removeNeutralParticles"),
+	  Comment("Removes steps of neutral particles"),true};
+      };
+      using Parameters = art::EDProducer::Table<Config>;
+      explicit CrvStepsFromStepPointMCs(const Parameters& conf);
+
+    private:
+      void beginJob() override;
+      void beginRun(art::Run& run) override;
+      void produce(art::Event& e) override;
+      typedef pair<CRSScintillatorBarIndex,cet::map_vector_key> BarIndexTrackPair; // key for pair of Crv barIndex, SimParticle
+      typedef art::Ptr<StepPointMC> SPMCPtr;
+      typedef vector<SPMCPtr> SPMCPtrV;
+      typedef map<BarIndexTrackPair, SPMCPtrV> StepPointMap; // steps by Crv barIndex and SimParticle
+      typedef art::Handle<StepPointMCCollection> SPMCCH;
+      typedef vector< SPMCCH > SPMCCHV;
+      void fillMap(SPMCCH const& spmcch, StepPointMap& stepPointMap, GlobalConstantsHandle<ParticleDataTable> &pdt);
+      void fillSteps(SPMCPtrV const& spmcptrv, CRSScintillatorBarIndex const& barIndex,
+	             ParticleData const* pdata, cet::map_vector_key trackId,
+                     unique_ptr<CrvStepCollection> &crvSteps, vector<pair<size_t,size_t> > &spmcIndices);
+      void fillStepDiag(CrvStep const& crvStep, SPMCPtrV const& spmcptrv, size_t firstIndex, size_t lastIndex);
+      static bool compareStepTimes(SPMCPtr a, SPMCPtr b) {return(a->time()<b->time());}
+
+      int      _debug, _diag;
+      bool     _removeNeutralParticles;
+      unsigned _csize, _ssize;
+      // StepPointMC selector
+      // This selector will select only data products with the given instance name.
+      // optionally exclude modules: this is a fix
+      art::Selector _selector;
+      bool _firstEvent;
+      // diagnostic histograms
+      TTree*          _csdiag;
+      Float_t         _cslen, _csdist, _csedep, _csPartP;
+      vector<Float_t> _slen, _sdist, _sedep, _sPartP;
+      Int_t           _csPartPDG;
+      vector<Int_t>   _sPartPDG;
+  };
+
+  CrvStepsFromStepPointMCs::CrvStepsFromStepPointMCs(const Parameters& config )  : 
+    art::EDProducer{config},
+    _debug(config().debug()),
+    _diag(config().diag()),
+    _removeNeutralParticles(config().removeNeutralParticles()),
+    _csize(config().csize()),
+    _ssize(config().startSize()),
+    _selector{art::ProductInstanceNameSelector(config().crvSteps()) &&
+      !art::ModuleLabelSelector(config().stepsToSkip()) },
```

Comment:
what is the use case for excluding some stepoint collections?

Reply:
Examples are the stage1 cosmic jobs and the cosmic validation script, where we have StepPointMCs with g4run and cosmicFilter module labels. Just before you wrote this comment, I pushed a change: The module is now asking for a list of StepPointMC labels (also accepting "*") to be used instead of excluding a StepPointMC label.